[ OEP 3904 ]: add Volume Group Snapshot to ZFS, LVM and Mayastor

tiagolobocastro opened the pull request:
This OEP proposes adding a Volume Group Snapshot feature to OpenEBS ZFS.
This feature would enable users to create consistent snapshots of multiple volumes in a single operation.
It is designed to improve data consistency across related volumes, particularly for stateful applications with interdependent datasets.

tiagolobocastro committed:
oep(3904): add volume-group-snapshot to mayastor, zfs and lvm

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

avishnu commented:
Thanks @tiagolobocastro for raising this. Indeed useful for ensuring nearline application-consistent backups.

1. Can we include localpv-lvm as well?
2. Shouldn't we include volume group restore?

tiagolobocastro commented:
> Thanks @tiagolobocastro for raising this. Indeed useful for ensuring nearline application-consistent backups.
> 
>     1. Can we include localpv-lvm as well?
> 
>     2. Shouldn't we include volume group restore?

AIUI there's no volume group restore, snapshots are restored with existing snapshot restore workflow.
I had left snapshot out of here since we don't have snapshot restore workflow for lvm, but I guess I can add it nonetheless.

tiagolobocastro committed:
build(nix-shell): remove linux specific utils from aarch-darwin

Allows us to run the nix-shell on darwin, simply in a way to build the
plugin and any non-linux dependent binaries.
The nvme-cli and kmod was used for testing on kind clusters, which is
assuming that the host is linux.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
chore: update oep issue template

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>
